### ISSUE TITLE
OOD data is positive or negative?

### ISSUE THREAD

**Issue by FrancescoCappio** [tayden/ood-metrics#2]
Some of the OOD metrics give different results depending on the definition of the positive class. This is for example the case of FPR at TPR 95 and AUPR. 

In the documentation of the function `calc_metrics` I see that this function should provide results according to the definition of the ODIN paper "Enhancing The Reliability Of Out-of-Distribution Image Detection In Neural Networks". At the same time the documentation reports that negative samples should receive a ground truth label of 1 (and correspondingly positive samples should get 0 or -1). 

These two definitions seem to be inconsistent to me:

 - in ODIN the authors clearly state that for FPR at 95% TPR the in-distribution (ID) samples are considered as positives whereas OOD samples are treated as negatives. As a result, for your code to compute their fpr_at_95_tpr score, the positive class should get ground truth label 1 (i.e. ID label = 1). 
 - in ODIN they use AUPR-IN for the case ID = positive and AUPR-OUT for the case OOD = positive. In your code you use original scores/labels to compute AUPR-OUT (expecting therefore negative samples to have label 1) and invert them to get AUPR-IN. 

The consequence is that if I understood everything correctly: 

 - if one calls `calc_metrics` with label 1 for OOD samples the AUPR-IN/AUPR-OUT results will be correct (matching ODIN definition), but the FPR at 95% TPR value will not match ODIN definition;
 - conversely if one calls `calc_metrics` with label 1 for ID samples the FPR at 95% TPR value will be correct (matching ODIN definition) while AUPR-IN and AUPR-OUT values will be inverted. 

Have I understood this correctly?

**Comment by tayden** [comment]
I think 2 things are at play here:

1. The documentation for the functions needs to be updated to say "ID" or "OOD" instead of "Negatives" and "Positives" to make expectations more clear.
2. It's possibly an oversight that I've flipped AUPR In and Out and they need to be reversed.

It's been quite a while since I've looked at this and OOD detection, but if you want to submit a PR with changes to improve the correctness of the library, I'd be happy to merge it in.
